src/operator/skipWhile.ts

```diff
@@ -38,17 +36,21 @@ class SkipWhileSubscriber<T> extends Subscriber<T> {
 
   protected _next(value: T): void {
     const destination = this.destination;
-    if (this.skipping === true) {
-      const index = this.index++;
-      const result = tryCatch(this.predicate)(value, index);
-      if (result === errorObject) {
-        destination.error(errorObject.e);
-      } else {
-        this.skipping = Boolean(result);
-      }
+    if (this.skipping) {
+      this.tryCallPredicate(value);
     }
-    if (this.skipping === false) {
+
+    if (!this.skipping) {
```

Comment:
Is the problem having the state, or updating it inside `tryCallPredicate` and check for its value on `_next()` right after? If the problem is the second, I agree with you that it's a bit confusing.

What about this?
```typescript
protected _next(value: T): void {
    const destination = this.destination;
    if (this.skipping) {
      this.skipping = this.tryCallPredicate(value);
    }

    if (!this.skipping) {
      destination.next(value);
    }
  }

  private tryCallPredicate(value: T): boolean {
    let skipping = true;
    try {
      const result = this.predicate(value, this.index++);
      skipping = Boolean(result);
    } catch (err) {
      this.destination.error(err);
    }
    return skipping;
  }
```